Cabal/Distribution/Simple/Utils.hs

```diff
@@ -235,23 +237,34 @@ dieWithLocation filename lineno msg =
 die :: String -> IO a
 die msg = ioError (userError msg)
 
-topHandlerWith :: (Exception.IOException -> IO a) -> IO a -> IO a
-topHandlerWith cont prog = catchIO prog handle
+topHandlerWith :: forall a. (Exception.SomeException -> IO a) -> IO a -> IO a
+topHandlerWith cont prog = Exception.catch prog handle
```

Comment:
Do we really want to catch all types of exception? I thought we did not, e.g. we do not want to catch the ctl-C exception. Is the purpose here to catch custom exception types from e.g. security failures?